[CI] UpdateHealthInfoCacheIT testMasterFailure failing

This has failed a few times in the last six months with the same failure message:

```
Expected: <DiskHealthInfo[healthStatus=GREEN, cause=null]>
     but: was null
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.health.UpdateHealthInfoCacheIT.assertResultsCanBeFetched(UpdateHealthInfoCacheIT.java:153)
		at org.elasticsearch.health.UpdateHealthInfoCacheIT.lambda$testHealthNodeFailOver$4(UpdateHealthInfoCacheIT.java:88)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1136)
		... 38 more
	Suppressed: java.lang.AssertionError: 
Expected: <DiskHealthInfo[healthStatus=GREEN, cause=null]>
     but: was null
		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
		at org.junit.Assert.assertThat(Assert.java:956)
		at org.junit.Assert.assertThat(Assert.java:923)
		at org.elasticsearch.health.UpdateHealthInfoCacheIT.assertResultsCanBeFetched(UpdateHealthInfoCacheIT.java:153)
		at org.elasticsearch.health.UpdateHealthInfoCacheIT.lambda$testHealthNodeFailOver$4(UpdateHealthInfoCacheIT.java:88)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1136)
		... 38 more
	Suppressed: java.lang.AssertionError: 
Expected: <DiskHealthInfo[healthStatus=GREEN, cause=null]>
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/atedejz57asri/tests/:server:internalClusterTest/org.elasticsearch.health.UpdateHealthInfoCacheIT/testMasterFailure

**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.health.UpdateHealthInfoCacheIT.testMasterFailure" -Dtests.seed=20B3752FEA453C10 -Dtests.locale=ro -Dtests.timezone=America/Antigua -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.health.UpdateHealthInfoCacheIT&tests.test=testMasterFailure

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <DiskHealthInfo[healthStatus=GREEN, cause=null]>
     but: was null

  at __randomizedtesting.SeedInfo.seed([20B3752FEA453C10:F859FCA181F45392]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.health.UpdateHealthInfoCacheIT.assertResultsCanBeFetched(UpdateHealthInfoCacheIT.java:131)
  at org.elasticsearch.health.UpdateHealthInfoCacheIT.lambda$testMasterFailure$5(UpdateHealthInfoCacheIT.java:103)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1172)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1145)
  at org.elasticsearch.health.UpdateHealthInfoCacheIT.testMasterFailure(UpdateHealthInfoCacheIT.java:103)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/health/UpdateHealthInfoCacheIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/health/UpdateHealthInfoCacheIT.java
@@ -24,6 +24,7 @@ import org.elasticsearch.test.junit.annotations.TestLogging;
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.notNullValue;
@@ -77,7 +78,7 @@ public class UpdateHealthInfoCacheIT extends ESIntegTestCase {
         DiscoveryNode newHealthNode = waitAndGetHealthNode(internalCluster);
         assertThat(newHealthNode, notNullValue());
         logger.info("Previous health node {}, new health node {}.", healthNodeToBeShutDown, newHealthNode);
-        assertBusy(() -> assertResultsCanBeFetched(internalCluster, newHealthNode, List.of(nodeIds), null));
+        assertBusy(() -> assertResultsCanBeFetched(internalCluster, newHealthNode, List.of(nodeIds), null), 30, TimeUnit.SECONDS);
     }
 
     @TestLogging(value = "org.elasticsearch.health.node:DEBUG", reason = "https://github.com/elastic/elasticsearch/issues/97213")
@@ -93,7 +94,7 @@ public class UpdateHealthInfoCacheIT extends ESIntegTestCase {
         ensureStableCluster(nodeIds.length);
         DiscoveryNode newHealthNode = waitAndGetHealthNode(internalCluster);
         assertThat(newHealthNode, notNullValue());
-        assertBusy(() -> assertResultsCanBeFetched(internalCluster, newHealthNode, List.of(nodeIds), null));
+        assertBusy(() -> assertResultsCanBeFetched(internalCluster, newHealthNode, List.of(nodeIds), null), 30, TimeUnit.SECONDS);
     }
 
     /**

